### PR TITLE
cmake: try system libraries first

### PR DESCRIPTION
**Pull request by abouvier**
Give package managers a chance :p

### CODE REVIEW

File: CMakeLists.txt
Comment (DHrpcs3) [inline CMakeLists.txt:12]: We need to verify that version of spirv-cross can parse SPIRV 1.6. Latest release of spirv-cross was in 2021 and it cannot parse newer extensions. We cannot determine which patches are applied by distributive maintainers by comparing versions because all will have same 2021 version

File: CMakeLists.txt
Comment (DHrpcs3) [inline CMakeLists.txt:17]: We will have fork of SPIRV-Tools with custom patches

**Review by DHrpcs3 — CHANGES_REQUESTED** [review]
Thank you for cleanup. Probably we can apply sox and unwind changes, but not SPIRV

File: CMakeLists.txt
Comment (abouvier) [reply CMakeLists.txt:12]: There is [many new tags](https://github.com/KhronosGroup/SPIRV-Cross/tags) that can be used. We can also create a `FindSPIRV-Cross.cmake` module that checks anything: function present in header, symbol present in shared library, or even a small program that try to parse a SPIRV 1.6 sample.

File: CMakeLists.txt
Comment (abouvier) [reply CMakeLists.txt:17]: Those patches cannot be made on the upstream repo? :p

File: CMakeLists.txt
Comment (DHrpcs3) [reply CMakeLists.txt:12]: You can try if you want :)

File: CMakeLists.txt
Comment (DHrpcs3) [reply CMakeLists.txt:17]: It doesn't matter, even after merge to upstream will need to wait almost year until patches available for most distros. But yes, I will make pull request to SPIRV-Tools

Comment (abouvier) [reply CMakeLists.txt:17]: Again, the new functionality can be tested and the system library rejected. The submodule will then be used. Bleeding edge distros should not be punished.

Comment (DHrpcs3) [reply CMakeLists.txt:17]: Sounds good, let's see how it will work

File: CMakeLists.txt
Comment (abouvier) [reply CMakeLists.txt]: Libraries not compatible with SPIRV 1.6 are now rejected ;)

**Review by DHrpcs3 — COMMENTED** [review]
OK, seems it works for me, thank you
Regarding SPIRV-Tools, could you please investigate how we can determine that latest 2 commits from following repository is included to precompiled library?
[https://github.com/RPCSX/SPIRV-Tools/commits/main/](https://github.com/RPCSX/SPIRV-Tools/commits/main/)

**Comment by abouvier** [comment]
> OK, seems it works for me, thank you Regarding SPIRV-Tools, could you please investigate how we can determine that latest 2 commits from following repository is included to precompiled library? https://github.com/RPCSX/SPIRV-Tools/commits/main/

I added a check for the presence of `spvtools::CreateBitCastCombinePass`, but the other commit does not change the public API. How can a program use the new functions?

**Comment by DHrpcs3** [comment]
> > OK, seems it works for me, thank you Regarding SPIRV-Tools, could you please investigate how we can determine that latest 2 commits from following repository is included to precompiled library? https://github.com/RPCSX/SPIRV-Tools/commits/main/
> 
> I added a check for the presence of `spvtools::CreateBitCastCombinePass`, but the other commit does not change the public API. How can a program use the new functions?

They shouldn't be used explicitly, it extends already exists optimization pipeline

**Comment by DHrpcs3** [comment]
Looks good now, thank you

**Pull request merged**
